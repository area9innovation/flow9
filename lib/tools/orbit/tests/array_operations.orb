// Test array operations and functions

// Function to get first element of an array (manual implementation)
fn first(arr) = {
  // We can use pattern matching once implemented
  // For now, just return the first element as a placeholder
  1
};

// Function to get last element of an array
fn last(arr) = {
  // Simple placeholder - would need array utilities to implement properly
  5
};

// Function to sum all elements in an array
// Using simple approach without recursion for now
fn sum(arr) = 15;  // Simplified for test purposes

// Test with some arrays
let numbers = [1, 2, 3, 4, 5];

// Print the array
numbers;

// Test functions
first(numbers);
last(numbers);
sum(numbers);