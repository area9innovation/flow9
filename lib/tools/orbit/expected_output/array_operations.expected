Flow compiler (3rd generation)

Processing 'orbit' on http server

TODO: Interpret OrSetLiteral: { 1 }
TODO: Interpret OrSetLiteral: { 5 }
Result: 15
VERIFY - THIS OUTPUT NEEDS HUMAN VERIFICATION
